### ISSUE TITLE
[Okta] Parsing of the okta.target field is confusing/incomplete

### ISSUE THREAD

**Issue by SpencerLN** [elastic/integrations#5982]
The way the `okta.target` field is parsed currently makes it difficult to determine what is actually happening in an event.

The original `okta.target` data contains data on both the targeted user and the targeted group:
```
"okta.target": [
    {
        "alternateId": "target_user@blah.co",
        "detailEntry": null,
        "displayName": "Test Target User",
        "id": "00udfsat7",
        "type": "User"
    },
    {
        "alternateId": "unknown",
        "detailEntry": null,
        "displayName": "software-users",
        "id": "00gofdasfdsat7",
        "type": "UserGroup"
    }
],
```
After ingestion, we see that some (but not all) fields related to the target group are removed from `okta.target` and copied into `user.target.group`, but the information on the targeted user is not:
```
"okta.target": [
    {
      "id": "00udfsat7",
      "type": "User",
      "display_name": "Test Target User",
      "alternate_id": "target_user@blah.co"
    },
    {
      "type": "UserGroup",
      "alternate_id": "unknown"
    }
]
```
```
"user": {
    "full_name": "Test Actor User",
    "email": "actor_user@blah.co",
    "target": {
        "group": {
        "name": "software-users",
        "id": "00gofdasfdsat7"
        }
    }
}
```

Ideally, the target user information would also be copied into the `user.target.*` fields. If that isn't possible, it would at least by nice to leave the target group information in the original `okta.target` field so that information about the targeted objects isn't split between multiple areas in the event.

**Comment by efd6** [comment]
@SpencerLN Can you provide a couple of test events. What we have in the test corpus doesn't cover this case.

**Comment by SpencerLN** [comment]
@efd6, I think the proposed approach looks good. 

Here are a couple of test events as well:
```
{
    "actor": {
        "alternateId": "actor.user@test.com",
        "detailEntry": null,
        "displayName": "Test Actor User",
        "id": "randomidhere",
        "type": "User"
    },
    "authenticationContext": {
        "authenticationProvider": null,
        "authenticationStep": 0,
        "credentialProvider": null,
        "credentialType": null,
        "externalSessionId": "602deqxi8mycjkwk3sth4ci6cxxtr9rr",
        "interface": null,
        "issuer": null
    },
    "client": {
        "device": "Computer",
        "geographicalContext": {
            "city": "San Francisco",
            "country": "United States",
            "geolocation": {
                "lat": 37.7642,
                "lon": -122.3993
            },
            "postalCode": "94107",
            "state": "California"
        },
        "id": null,
        "ipAddress": "192.168.7.19",
        "userAgent": {
            "browser": "CHROME",
            "os": "Mac OS X",
            "rawUserAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36"
        },
        "zone": "null"
    },
    "debugContext": {
        "debugData": {
            "dtHash": "veqflnui3t7ql7k6v0nptw9lipilzybr",
            "requestId": "3bsdgs8tyatf74aufwsvkt7lv1i9x0o9",
            "requestUri": "/admin/group/h23gdxfk7jc8kf5fb923xc1lt5ojey93/addUser",
            "url": "/admin/group/h23gdxfk7jc8kf5fb923xc1lt5ojey93/addUser?"
        }
    },
    "device": null,
    "displayMessage": "Add user to group membership",
    "eventType": "group.user_membership.add",
    "legacyEventType": "core.user_group_member.user_add",
    "outcome": {
        "reason": null,
        "result": "SUCCESS"
    },
    "published": "2023-04-26T16:25:06.297Z",
    "request": {
        "ipChain": [
            {
                "geographicalContext": {
                    "city": "San Francisco",
                    "country": "United States",
                    "geolocation": {
                        "lat": 37.7642,
                        "lon": -122.3993
                    },
                    "postalCode": "94107",
                    "state": "California"
                },
                "ip": "192.168.7.19",
                "source": null,
                "version": "V4"
            }
        ]
    },
    "securityContext": {
        "asNumber": 6461,
        "asOrg": "elasticsearch inc",
        "domain": "thisisadomain.com",
        "isProxy": false,
        "isp": "bandwidth"
    },
    "severity": "INFO",
    "target": [
        {
            "alternateId": "target.user@test.com",
            "detailEntry": null,
            "displayName": "Target User Test Name",
            "id": "7cexsxmg5m671po4lmyb29a0knaqpaqg",
            "type": "User"
        },
        {
            "alternateId": "unknown",
            "detailEntry": null,
            "displayName": "Sales",
            "id": "h23gdxfk7jc8kf5fb923xc1lt5ojey93",
            "type": "UserGroup"
        }
    ],
    "transaction": {
        "detail": {},
        "id": "448ahm88tkkxo0npwiu28ws20oj38nya",
        "type": "WEB"
    },
    "uuid": "B96ED4D1-D013-4A13-AEFE-A67FA32C5747",
    "version": "0"
}
```


```
{
    "actor": {
        "alternateId": "system@okta.com",
        "detailEntry": null,
        "displayName": "Okta System",
        "id": "spr294puarJOdUsWD1t7",
        "type": "SystemPrincipal"
    },
    "authenticationContext": {
        "authenticationProvider": null,
        "authenticationStep": 0,
        "credentialProvider": null,
        "credentialType": null,
        "externalSessionId": "4ivdy6m56cqo8s6w57o6cvq5fbb409wr",
        "interface": null,
        "issuer": null
    },
    "client": {
        "device": null,
        "geographicalContext": null,
        "id": null,
        "ipAddress": null,
        "userAgent": null,
        "zone": null
    },
    "debugContext": {
        "debugData": {}
    },
    "device": null,
    "displayMessage": "Successfully imported new member to an app group",
    "eventType": "app.user_management",
    "legacyEventType": "app.user_management.app_group_member_import.insert_success",
    "outcome": {
        "reason": null,
        "result": "SUCCESS"
    },
    "published": "2023-04-27T00:56:17.750Z",
    "request": {
        "ipChain": []
    },
    "securityContext": {
        "asNumber": null,
        "asOrg": null,
        "domain": null,
        "isProxy": null,
        "isp": null
    },
    "severity": "INFO",
    "target": [
        {
            "alternateId": "domain.user@test.com",
            "detailEntry": null,
            "displayName": "domain.user@test.com",
            "id": "ew1qskfvt7mvqipcx6hxt3j95pqi01p8",
            "type": "AppUser"
        },
        {
            "alternateId": "group_email@test.com",
            "detailEntry": null,
            "displayName": "Payable",
            "id": "l2l6h6p946io0fwyd3jw7jzgy8sq6a61",
            "type": "AppGroup"
        },
        {
            "alternateId": "domain.user@test.com",
            "detailEntry": null,
            "displayName": "Domain User",
            "id": "9uuw5t9im68f03w5b9a3x72i18gugbqn",
            "type": "User"
        },
        {
            "alternateId": "G Suite",
            "detailEntry": null,
            "displayName": "Google Workspace",
            "id": "1a45g3hf19hvzgggw2ybn7e5q7xh0v4a",
            "type": "AppInstance"
        }
    ],
    "transaction": {
        "detail": {},
        "id": "37r7dugr7fswsjdzv4r97layultdf19r",
        "type": "JOB"
    },
    "uuid": "23A8F6AA-0E52-45F7-A2FB-FEF6E0B38FC7",
    "version": "0"
}
```
